src/common/state/redis.c

```diff
@@ -399,12 +432,17 @@ void object_table_redis_callback(redisAsyncContext *c,
   CHECK(reply->elements > 2);
   /* If this condition is true, we got the initial message that acknowledged the
    * subscription. */
-  if (strncmp(reply->element[1]->str, "add", 3) != 0) {
-    if (callback_data->done_callback) {
-      object_table_done_callback done_callback = callback_data->done_callback;
-      done_callback(callback_data->id, callback_data->user_context);
+  bool is_add = reply->element[1]->str && strcmp(reply->element[1]->str, "add");
+  if (is_add) {
+    /* Do a lookup to see if the key has been in redis before we started the
+     * subscription. */
+    int status = redisAsyncCommand(
```

Comment:
wouldn't it make sense to parse the payload first, and only do the SMEMBERS command if the object table does not contain any managers?